consensus/src/marshal/config.rs

```diff
@@ -0,0 +1,80 @@
+use crate::Block;
+use commonware_cryptography::{bls12381::primitives::variant::Variant, PublicKey};
+use commonware_resolver::p2p::Coordinator;
+use governor::Quota;
+
+/// Marshal configuration.
+#[derive(Debug)]
+pub struct Config<V: Variant, P: PublicKey, Z: Coordinator<PublicKey = P>, B: Block> {
+    /// The public key of the validator.
+    pub public_key: P,
+
+    /// The identity of the network.
+    pub identity: V::Public,
+
+    /// The coordinator for the resolvers.
+    pub coordinator: Z,
+
+    /// The prefix to use for all partitions.
+    pub partition_prefix: String,
+
+    /// Size of backfill request/response mailbox.
+    pub mailbox_size: usize,
+
+    /// Backfill rate limit.
+    pub backfill_quota: Quota,
+
+    /// Minimum grace period for retaining activity after the application has processed the block.
+    ///
+    /// For notarizations, this is in number of views.
+    /// For blocks and finalizations, this is in number of blocks.
+    ///
+    /// Useful for keeping around information that peers may desire to have.
+    pub grace_period: u64,
```

Comment:
👍

Reply:
Called it `view_retention_timeout` in https://github.com/commonwarexyz/monorepo/pull/1306/commits/3b35a0ba9e993ef0c0a824e73227b4ffc5cd5874